### ISSUE TITLE
highlight development.html as part of the guide ~2

### ISSUE THREAD

**Issue by robertheessels** [donejs/donejs#519]
We should tell people about development.html in Chat and PMO once they get things setup.

Original issue:

> I have been using CanJs for many years. Now following the DoneJs guides was a breeze. After that I created a brand new DoneJs app, but pretty soon discovered autorender was not the way to go for this project.
> 
> I could not find out however, even after MANY hours of searching and trying, how to just start the app in the browser WITHOUT autorender (after running "donejs develop"). Just like you would do with an old school CanJs app.
> 
> Turns out that after "donejs develop" you can just hit http://localhost:8080/development.html
> and you then skipped autorender and just run your app like you used to with CanJs.
> 
> Please add another guide explaining all this asap! :-)
> 
> By the way, I am VERY impressed with DoneJs, even after using CanJs for many years. great job, guys. You rock!!!

**Comment by robertheessels** [comment]
It is also unclear how and when to use "import" or "steal()" when you run your app this way.

And it seems like I have to do some check if I'm running in node like this:

``` javascript
var ViewModel = Map;
if( typeof module !== 'undefined' && module.exports ) {
    // On server side do not set the viewModel, because it does not seem to be able to handle
    // 401 (unauthorized) 'errors'.
} else {
    // Only load the viewModel when on client side.

    ViewModel = Map.extend( {
        define: {
            login_status: {
                value: 0
            },

            threads: {
                value() {
                    return Thread.getList( {} );
                }
            },

            me: {
                value() {
                    var self = this;
                    var dfd = new $.Deferred();

                    wait( function() {
                        Myself.get( { id: 'me' } ).then( function( data ) {
                            dfd.resolve( data );
                            self.attr( 'login_status', 1 );
                        }, function() {
                            self.attr( 'login_status', 401 );
                        } );
                    } );

                    return dfd;
                }
            }
        }
    } );
}

export default Component.extend( {
    tag: 'tdd-home',
    viewModel: ViewModel,
    template
} );
```

or I get this warning and the component will not show:

WARN: can/view/scanner.js: No custom element found for tdd-home

This seems to be related with running client or server side?

**Comment by pYr0x** [comment]
you want to use can-ssr or not?

**Comment by justinbmeyer** [comment]
@robertheessels I don't think this is going to be a guide on DoneJS.com.  `development.html` shouldn't even really be used if you have pushstate active.  DoneJS is for the combination of technologies.  The subprojects should have guides when you aren't using the full package.

What are you trying to setup?  Just CanJS+Steal?  If that's the case, what more do you need besides: http://canjs.com/guides/Using.html#section_StealJSandNPM ?

I'm not sure `done-autorender` is appropriate for non-server-side rendered code.  The code in CanJS's place-my-order guide should be sufficient.  

> It is also unclear how and when to use "import" or "steal()" when you run your app this way.

I'm not sure exactly which way you are running your app.  But no matter how you are running it, you shouldn't have code like:

```
if( typeof module !== 'undefined' && module.exports ) {
    // On server side do not set the viewModel, because it does not seem to be able to handle
    // 401 (unauthorized) 'errors'.
} else {
```

**Comment by robertheessels** [comment]
I want to use as much of DoneJs as possible. And I want the app to run on the web, and as standalone apps (Android etc).

As soon as my app starts, it needs to check if the user is authenticated, based on a token. If not, a 401 is returned by the REST server.

When I try to catch that 401 without a check for module.exports and without wait() (as can be seen in my previous comment) the app fails to run. Not sure if that is a bug in DoneJs.

So... because I have to check the authentication right away, there is not much use in rendering server-side? Or do I miss something?

But still I want to be able to build for standalone apps etc.

**Comment by phillipskevin** [comment]
Others have had a similar question about how to run a DoneJS app without server-side rendering. Maybe just an addition to https://github.com/donejs/generator-donejs about what files are created when you generate a new app?

**Comment by phillipskevin** [comment]
But it might help to add that to http://donejs.com/Guide.html#section=section_Generatetheapplication also. Not sure where the best place is.

**Comment by robertheessels** [comment]
In the in depth guide it is explained what files are generated and what they are for. So that is fine, I guess. http://donejs.com/place-my-order.html

What is unclear to me is how is decided what runs server side and what runs client side. How does that work? How can you know?

I have now installed auth-component (https://github.com/marshallswain/auth-component) and that solved my problem. Somehow it seems to run client side only, but I do not yet understand why it does I my own code that does the same, but via e Supermap doesn't.

So a manual explaining the dynamics of how server side and client side 'cooperate' would be awesome.

**Comment by marshallswain** [comment]
@robertheessels, I'm surprised you found that, but I'm glad it helped.  If I remember correctly, I kept it from running on the server by checking for window.localStorage (which I noticed doesn't exist on the server) in the `init` [here](https://github.com/marshallswain/auth-component/blob/master/src/token-auth/token-auth.js#L203)

It's actually on my agenda to update that package rather extensively in the next couple of weeks and really give it some polish.  Just a heads up, the API will definitely change.  Right now it's more of a proof of concept that works well enough to use.  Please create issues for anything you'd like to see happen.

**Comment by robertheessels** [comment]
I just copied parts of your code in my app and got everything working finally. So I have no need for future versions.

Unfortunately it didn't help me understand the dynamics between sever- and client side.

Having said that, it certainly has potential, Could be an important package for DoneJs because, I guess, most projects need authentication.

**Comment by marshallswain** [comment]
Ok, good.  That way I won't be messing you up with updates.  :)  Cheers!

**Comment by justinbmeyer** [comment]
@James0x57 once the cookie stuff lands, maybe you can point @robertheessels at how SSR works with authentication.

**Comment by matthewp** [comment]
Is there anything actionable to do here? I understand that some people will not to use server-side rendering, but I'm not sure where that should be documented. How about a form post and then link to that from the guide?

**Comment by justinbmeyer** [comment]
This probably should be documented as part of `donejs add app`.  We should link to it from at least the PMO guide.

**Comment by matthewp** [comment]
I documented this [here](https://github.com/donejs/generator-donejs#developmenthtml) and linked to it in the PMO guide.
